cdap-common/src/test/java/co/cask/cdap/io/SchemaTest.java

```diff
@@ -149,6 +153,43 @@ public void testSchemaHash() throws UnsupportedTypeException {
   }
 
   @Test
+  public void testAvroEnumSchema() throws Exception {
+    org.apache.avro.Schema schema = org.apache.avro.Schema.createEnum("UserInterests", "Describes interests of user",
+                                                                        "org.example.schema",
+                                                                        ImmutableList.of("CRICKET", "BASEBALL"));
+
+    Schema.parseJson(schema.toString());
+  }
+
+  @Test
+  public void testAvroRecordSchema() throws Exception {
+
+    org.apache.avro.Schema avroStringSchema = org.apache.avro.Schema.create(org.apache.avro.Schema.Type.STRING);
+    org.apache.avro.Schema avroIntSchema = org.apache.avro.Schema.create(org.apache.avro.Schema.Type.INT);
+
+
+    org.apache.avro.Schema schema = org.apache.avro.Schema.createRecord("UserInfo", "Describes user information",
+                                                                        "org.example.schema", false);
+
+    List<org.apache.avro.Schema.Field> fields = new ArrayList<>();
+
+    org.apache.avro.Schema.Field field = new org.apache.avro.Schema.Field("username", avroStringSchema,
+                                                                          "Field represents username",
+                                                                          new TextNode("unknown"));
+    fields.add(field);
+
+
+    field = new org.apache.avro.Schema.Field("age", avroIntSchema, "Field represents age of user",
+                                             new IntNode(-1));
+
+    fields.add(field);
+
+    schema.setFields(fields);
+
+    Schema.parseJson(schema.toString());
```

Comment:
would it be good to actually assert the schema instead of just relying on the ``parseJson`` to not error out?